Spigot-API-Patches/0296-Add-missing-MemoryKeys.patch

```diff
@@ -0,0 +1,128 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Sun, 9 May 2021 19:35:09 -0700
+Subject: [PATCH] Add missing MemoryKeys
+
+
+diff --git a/src/main/java/org/bukkit/entity/memory/MemoryKey.java b/src/main/java/org/bukkit/entity/memory/MemoryKey.java
+index 905d5f6d16a6e3f8eb9da91cbc3ba3286ad15e4b..c92042b683e783a0f0db3629a9925b6691ea5c3b 100644
+--- a/src/main/java/org/bukkit/entity/memory/MemoryKey.java
++++ b/src/main/java/org/bukkit/entity/memory/MemoryKey.java
+@@ -10,6 +10,19 @@ import org.bukkit.Location;
+ import org.bukkit.NamespacedKey;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
++// Paper start
++import org.bukkit.entity.Hoglin;
++import org.bukkit.entity.Mob;
++import org.bukkit.entity.PiglinAbstract;
++import java.util.List;
++import com.google.common.reflect.TypeToken;
++import org.bukkit.entity.LivingEntity;
++import org.bukkit.entity.Ageable;
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.HumanEntity;
++import org.bukkit.entity.Item;
++import static org.bukkit.NamespacedKey.minecraft;
++// Paper end
+ 
+ /**
+  * Represents a key used for accessing memory values of a
+@@ -21,12 +34,22 @@ public final class MemoryKey<T> implements Keyed {
+ 
+     private final NamespacedKey namespacedKey;
+     private final Class<T> tClass;
++    private final TypeToken<T> typeToken; // Paper
```

Comment:
I'd prefer we use https://github.com/leangen/geantyref for this - it is a new library, but other platforms (sponge, configurate) have moved away from guava's TypeToken as well.

/cc @zml2008

Reply:
Ok, switched it geantyref